daemon/graphdriver/overlay2/overlay.go

```diff
@@ -266,7 +266,7 @@ func supportsOverlay() error {
 
 func useNaiveDiff(home string) bool {
 	useNaiveDiffLock.Do(func() {
-		if err := hasOpaqueCopyUpBug(home); err != nil {
+		if err := hasOpaqueCopyUpBugOrEnableRedirectDir(home); err != nil {
```

Comment:
Ok. I was not aware of `sync.Once`. So there must have been some reason that we could not do in Init() and had to rely on this sync.Once stuff.